src/content/docs/en/guides/testing.mdx

```diff
@@ -16,6 +16,115 @@ A Vite-native unit test framework with ESM, TypeScript and JSX support powered b
 
 See the [Astro + Vitest starter template](https://github.com/withastro/astro/tree/latest/examples/with-vitest) on GitHub.
 
+## Cypress
+
+Cypress is a next generation front end testing tool built for the modern web. Cypress enables you to write all types of tests:
+
+* End-to-end tests
+* Component tests
+* Integration tests
+* Unit tests
+
+Cypress can test anything that runs in a browser.
+
+### Installation
+
+You can install Cypress within your Astro project using the package manager of your choice.
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+  ```shell
+  npm install -D cypress
+  ```
+
+  This will install Cypress locally as a dev dependency for your project.
+  </Fragment>
+  <Fragment slot="pnpm">
+  ```shell
+  pnpm add cypress --save-dev
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  yarn add cypress --dev
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+### Configuration
+
+In the root of your project, create a `cypress.config.ts` file with the following content:
+
+```json
+import { defineConfig } from 'cypress'
+
+export default defineConfig({
+  e2e: {
+    supportFile: false
+  },
+})
+```
+
+### Create your first Cypress test
+
+1. Choose a page to test. We will use the example `index.astro` page below.
+
+    ```html title="src/pages/index.astro"
+    ---
+    ---
+    <html lang="en">
+      <head>
+        <title>Astro is awesome!</title>
+        <meta name="description" content="Pull content from anywhere and serve it fast with Astro's next-gen island architecture." />
+      </head>
+      <body>
+      <h1>Hello world from Astro</h1>
+      </body>
+    </html>
+    ```
+
+2. Create an `index.spec.js` file in the `cypress/e2e` folder. Copy and paste the following test into the file to verify that the page title is correct. Update the value of the page `<title>` to match the page you are testing. 
+
+    ```js title="cypress/e2e/index.cy.ts"
+    it('titles are correct', () => {
+      const page = cy.visit('http://localhost:3000');
+
+      page.get('title').should('have.text', 'Astro is awesome!')
+      page.get('h1').should('have.text', 'Hello world from Astro');
+    });
+    ```
+
+    :::tip[set a baseUrl]
+    You can set [`"baseUrl": "http://localhost:3000"`](https://docs.cypress.io/guides/end-to-end-testing/testing-your-app#Step-3-Configure-Cypress) in the `cypress.config.ts` configuration file to use `cy.visit("/")` instead of `page.goto("http://localhost:3000/")` for a more convenient URL.
+    :::
+
+### Running your Cypress tests
+
+You can run a single test or several tests at once, testing one or multiple browsers. By default, Cypress uses a built-in instance of [Electron](https://www.electronjs.org/) to run your tests, and it also [supports](https://docs.cypress.io/guides/guides/launching-browsers) Firefox and Chrome-family browsers (including Edge).
```

Comment:
If we're not explaining _how_ to run several tests at once and how to test in multiple browsers, let's remove this paragraph